src/main/java/com/aws/greengrass/deployment/activator/KernelUpdateActivator.java

```diff
@@ -58,13 +58,18 @@ public void activate(Map<String, Object> newConfig, Deployment deployment,
         if (!takeConfigSnapshot(totallyCompleteFuture)) {
             return;
         }
-
-        if (!kernelAlternatives.isLaunchDirSetup()) {
+        try {
+            kernelAlternatives.validateLaunchDirSetupVerbose();
+        } catch (DirectoryValidationException e) {
+            logger.atError().setCause(e).log("Failed to validate launch directory");
             totallyCompleteFuture.complete(
                     new DeploymentResult(DeploymentResult.DeploymentStatus.FAILED_NO_STATE_CHANGE,
                             new DeploymentException("Unable to process deployment. Greengrass launch directory"
-                                    + " is not set up or Greengrass is not set up as a system service",
-                                    DeploymentErrorCode.LAUNCH_DIRECTORY_CORRUPTED)));
+                                    + " is not set up or Greengrass is not set up as a system service", e)));
+            return;
+        } catch (DeploymentException e) {
+            totallyCompleteFuture.complete(
```

Comment:
log here?

Or else if we log somewhere else (which seems likely), then don't log on L64 above